styx-common/src/main/java/com/spotify/styx/storage/BigtableStorage.java

```diff
@@ -107,29 +109,46 @@ void writeEvent(SequenceEvent sequenceEvent) throws IOException {
     });
   }
 
-  List<WorkflowInstanceExecutionData> executionData(WorkflowId workflowId)
+  List<WorkflowInstanceExecutionData> executionData(WorkflowId workflowId, String offset, int limit)
       throws IOException {
     final Table eventsTable = connection.getTable(EVENTS_TABLE_NAME);
 
     final Scan scan = new Scan()
         .setRowPrefixFilter(Bytes.toBytes(workflowId.toKey() + '#'))
         .setFilter(new FirstKeyOnlyFilter());
 
-    final Set<WorkflowInstance> workflowInstancesSet = Sets.newHashSet();
-    for (Result result : eventsTable.getScanner(scan)) {
-      final String key = new String(result.getRow());
-      final int lastHash = key.lastIndexOf('#');
-      final WorkflowInstance wfi = WorkflowInstance.parseKey(key.substring(0, lastHash));
-      workflowInstancesSet.add(wfi);
+    if (!Strings.isNullOrEmpty(offset)) {
+      final WorkflowInstance offsetInstance = WorkflowInstance.create(workflowId, offset);
+      scan.setStartRow(Bytes.toBytes(offsetInstance.toKey() + '#'));
```

Comment:
Will this include the item named in `offset` or start from the next one? It's not immediately obvious to me.